test/helpers.ts

```diff
@@ -1,24 +1,37 @@
 import * as rpc from "vscode-jsonrpc"
 import { TextEditor } from "atom"
+import AutoLanguageClient from "../lib/auto-languageclient"
+import { LanguageClientConnection, Disposable, Event } from "../lib/languageclient"
+import { LanguageServerProcess } from "../lib/server-manager"
+import { spawn } from "spawk"
+
+function createSpyDisposable(): Disposable {
+  return { dispose: jasmine.createSpy("dispose") }
+}
+
+// eslint-disable-next-line @typescript-eslint/ban-types
+function createSpyEvent(): Event<Disposable> {
+  return jasmine.createSpy("event").and.returnValue(createSpyDisposable())
+}
 
 export function createSpyConnection(): rpc.MessageConnection {
   return {
-    listen: jasmine.createSpy("listen"),
-    onClose: jasmine.createSpy("onClose"),
-    onError: jasmine.createSpy("onError"),
-    onDispose: jasmine.createSpy("onDispose"),
-    onUnhandledNotification: jasmine.createSpy("onUnhandledNotification"),
-    onUnhandledProgress: jasmine.createSpy("onUnhandledProgress"),
-    onRequest: jasmine.createSpy("onRequest"),
-    onNotification: jasmine.createSpy("onNotification"),
-    onProgress: jasmine.createSpy("onProgress"),
-    dispose: jasmine.createSpy("dispose"),
-    sendRequest: jasmine.createSpy("sendRequest"),
-    sendNotification: jasmine.createSpy("sendNotification"),
-    sendProgress: jasmine.createSpy("sendProgress"),
-    trace: jasmine.createSpy("trace"),
-    inspect: jasmine.createSpy("inspect"),
-    end: jasmine.createSpy("end"),
+    listen: jasmine.createSpy("listen").and.returnValue(void 0),
```

Comment:
why return `void 0` it should already return `undefined` by default.

Reply:
That's for clarity as it explicitly shows the intention of the spy.